test/e2e/workflow/deployment_test.go

```diff
@@ -454,6 +457,123 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			})
 		})
 	})
+
+	Context("NetworkAddonsConfig labels", func() {
+		var (
+			configSpec   cnao.NetworkAddonsConfigSpec
+			ovsDaemonSet = &v1.DaemonSet{}
+			ovsPods      = &corev1.PodList{}
+		)
+
+		BeforeEach(func() {
+			configSpec = cnao.NetworkAddonsConfigSpec{
+				Ovs: &cnao.Ovs{},
+			}
+
+			CreateConfig(gvk, configSpec)
+			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
+		})
+
+		Context("When NetworkAddonsConfig is labeled with `part-of`, `version` and `component` labels", func() {
+			const (
+				partOfLabelCRVal     = "part-of-cr-value"
+				versionLabelCRVal    = "version-cr-value"
+				componentLabelCRlVal = "component-cr-value"
+			)
+			labels := map[string]string{names.PART_OF_LABEL_KEY: partOfLabelCRVal,
+				names.VERSION_LABEL_KEY: versionLabelCRVal, names.COMPONENT_LABEL_KEY: componentLabelCRlVal}
+
+			BeforeEach(func() {
+				LabelConfig(gvk, labels)
+
+				Eventually(func() bool {
+					config := GetConfig(gvk)
+					configLabels := config.GetLabels()
+					return configLabels[names.PART_OF_LABEL_KEY] == partOfLabelCRVal && configLabels[names.VERSION_LABEL_KEY] == versionLabelCRVal &&
+						configLabels[names.COMPONENT_LABEL_KEY] == componentLabelCRlVal
+				}, 2*time.Minute, time.Minute).Should(BeTrue(), "should succeed setting labels on NetworkAddonsConfig")
+			})
+
+			It("Should take the `part-of`, `version` and `component` labels value and apply it to the deployed component daemonset", func() {
```

Comment:
in daemonsets and deployment we need to check the spec.labels and the spec.template.spec.labels. this way there is no need to check the actual pod

I would remove the pods check and add the deployemtn one